temporal-sdk/src/main/java/io/temporal/workflow/Workflow.java

```diff
@@ -593,26 +598,65 @@ public static ExternalWorkflowStub newUntypedExternalWorkflowStub(String workflo
   /**
    * Continues the current workflow execution as a new run with the same options.
    *
+   * <p> This method never returns control to the calling code.
+   *
    * @param args arguments of the next run.
    * @see #newContinueAsNewStub(Class)
    */
   public static void continueAsNew(Object... args) {
-    Workflow.continueAsNew(Optional.empty(), Optional.empty(), args);
+    Workflow.continueAsNew(null, args);
+  }
+
+  /**
+   * Continues the current workflow execution as a new run with the same workflowType and overridden
+   * {@code options}.
+   *
+   * <p> This method never returns control to the calling code.
+   *
+   * @param options option overrides for the next run, can contain null if no overrides are needed
+   * @param args arguments of the next run.
+   * @see #newContinueAsNewStub(Class, ContinueAsNewOptions)
+   */
+  public static void continueAsNew(@Nullable ContinueAsNewOptions options, Object... args) {
+    Workflow.continueAsNew(null, options, args);
   }
 
   /**
    * Continues the current workflow execution as a new run possibly overriding the workflow type and
    * options.
    *
-   * @param options option overrides for the next run.
+   * <p> This method never returns control to the calling code.
+   *
+   * @param workflowType workflow type override for the next run, can be null of no override is
+   *     needed
+   * @param options option overrides for the next run, can be null if no overrides are needed
    * @param args arguments of the next run.
    * @see #newContinueAsNewStub(Class)
    */
   public static void continueAsNew(
```

Comment:
The old method
`public static void continueAsNew(
      Optional<String> workflowType, Optional<ContinueAsNewOptions> options, Object... args)`
doesn't support nulls, so nobody should have such a code.

New code like that will lead to ambiguity, true. Good catch, BTW. I hope to get away with that, because nobody should be doing it this way anyway because there is a version of the method without workflowType and options and people should be using that version then.